hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-web-proxy/src/test/java/org/apache/hadoop/yarn/server/webproxy/TestWebAppProxyServerFed.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.hadoop.yarn.server.webproxy;
+
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+
+public class TestWebAppProxyServerFed extends TestWebAppProxyServer {
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    conf.setBoolean(YarnConfiguration.FEDERATION_ENABLED, true);
```

Comment:
Do we want to extend some tests just for federation or TestWebAppProxyServer is now generic enough?

Reply:
It's only for test the basic function (For example: start or stop and so on) of WebAppProxyServer in federation mode.

For me, I think TestWebAppProxyServer is not generic enough, because that UT only test start, stop and bindaddress. I don't know whether is necessary to add some integration testing so that we can test proxy app web link?

Reply:
@goiri I remove TestWebProxyServerFed. It is unnecessary for us to test startup. I add new UT TestWebAppProxyServletFed, will test proxy link in federation mode.